app/models/role.rb

```diff
@@ -74,6 +74,8 @@ def self.prime_minister_role
 
   def republish_associated_editions_to_publishing_api
     edition_roles.each do |edition_role|
+      next unless edition_role.edition
```

Comment:
Might be worth adding to the commit message an explanation that the reason this guard clause filters out edition roles with deleted editions is because the default scope of `Edition` excludes editions with a state of `deleted` (if my understanding is correct): https://github.com/alphagov/whitehall/blob/main/app/models/concerns/edition/workflow.rb#L21

Reply:
Done 👍